src/librustc/session/mod.rs

```diff
@@ -336,6 +336,10 @@ impl NodeIdAssigner for Session {
     fn peek_node_id(&self) -> NodeId {
         self.next_node_id.get().checked_add(1).unwrap()
     }
+
+    fn diagnostic(&self) -> &errors::Handler {
+        self.diagnostic()
```

Comment:
Um. This looks... weird. Did I introduce an infinite recursion (I didn't see a warning) or it's OK because inherent fns take precedence over trait fns?